tests/index_selector.json

```diff
@@ -85,6 +113,26 @@
       "selector": "$[01]",
       "invalid_selector": true
     },
+    {
+      "name": "decimal",
+      "selector": "$[1.0]",
+      "invalid_selector": true
+    },
+    {
+      "name": "plus",
+      "selector": "$[+1]",
+      "invalid_selector": true
+    },
+    {
+      "name": "minus space",
+      "selector": "$[- 1]",
```

Comment:
I think an argument could be made for this to be put in the whitespace tests as well, but I can see it staying here. 

Other opinions welcome.

Reply:
I'm getting more and more inclined to think we can benefit from tags. I might be able to make an attempt to add tags in a day or two.

Reply:
Yeah, I think a lot of these in particular could be categorized as just "parsing".  I don't think that it's practical to have 2^53 items in a JSON file 😆